src/components/@shared/FormInput/InputElement/ConsumerParameters/ConsumerParameterInput.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { ReactElement } from 'react'
+import Input, { InputProps } from '../..'
+import { Field, useField, useFormikContext } from 'formik'
+import {
+  FormConsumerParameter,
+  FormPublishData
+} from '@components/Publish/_types'
+import { defaultConsumerParam } from '.'
+
+export default function ConsumerParameterInput({
+  index,
+  inputName,
+  ...props
+}: InputProps & {
+  index: number
+  inputName: string
+}): ReactElement {
+  const { setFieldTouched } = useFormikContext<FormPublishData>()
+  const [field, meta, helpers] = useField<FormConsumerParameter[]>(inputName)
+
+  const resetDefaultValue = (
+    parameterName: string,
+    parameterType: FormConsumerParameter['type'],
+    index: number
+  ) => {
+    if (parameterName !== 'type') return
+    if (field.value[index].type === parameterType) return
+
+    setFieldTouched(`${field.name}[${index}].default`, false)
+    helpers.setValue(
+      field.value.map((currentParam, i) => {
+        if (i !== index) return currentParam
+
+        return {
+          ...defaultConsumerParam,
+          ...currentParam,
+          default: defaultConsumerParam.default
+        }
+      })
+    )
+  }
```

Comment:
thought: Reading over this file multiple times now I feel like it is probably best for readability to split this up logically
this function is only needed for the `type` prop basically (see below for another possible extraction)

So if this logic would be handled in a specific `TypeInput` component it is clearer on what the purpose might be.